File: Excel/425 Challenge.R
library(tidyverse)
library(readxl)

input = read_excel("Excel/425 Hex Color Blending.xlsx", range = "A1:B10")
test  = read_excel("Excel/425 Hex Color Blending.xlsx", range = "C1:C10")

result = input %>%
  mutate(Color1 = strsplit(as.character(input$Color1), ", ") %>%
           map(., ~as.numeric(.x)),
         Color2 = strsplit(as.character(input$Color2), ", ") %>%
           map(., ~as.numeric(.x))) %>%
  mutate(FinalColor = map2(Color1, Color2, ~ceiling((.x + .y) / 2))) %>%
  mutate(`Answer Expected` = map_chr(FinalColor, ~rgb(.x[1], .x[2], .x[3], maxColorValue = 255))) %>%
  select(-Color1, -Color2, -FinalColor)

identical(result, test)
#> [1] TRUE

File: Excel/400-499/425/425 Challenge.py
import pandas as pd

input = pd.read_excel("425 Hex Color Blending.xlsx",usecols="A:B", nrows = 10)
test = pd.read_excel("425 Hex Color Blending.xlsx", usecols="C", nrows = 10)

def split_and_convert_colors(color):
    return [int(c) for c in color.split(", ")]

input['Color1'] = input['Color1'].apply(split_and_convert_colors)
input['Color2'] = input['Color2'].apply(split_and_convert_colors)
input['FinalColor'] = input.apply(lambda row: [(c1 + c2 + 1) // 2 for c1, c2 in zip(row['Color1'], row['Color2'])], axis=1)
input['Answer Expected'] = input['FinalColor'].apply(lambda color: '#' + ''.join([hex(c)[2:].zfill(2).upper() for c in color]))
input = input.drop(columns=['Color1', 'Color2', 'FinalColor'])

print(input.equals(test)) # True
